orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/model/StageDefinition.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model;
+
+import java.util.List;
+import java.util.Map;
+
+public class StageDefinition implements NamedContent {
+
+  private String id;
+  private InjectionRule inject;
+  private String dependsOn; // TODO rz - comma-delimited for now
+  private String type;
+  private Map<String, Object> config;
+  private List<Map<String, Object>> notifications;
+  private String comments;
+
+  @Override
+  public String getName() {
+    return id;
+  }
+
+  public static class InjectionRule {
```

Comment:
I'm not 100% sure I understand the concept of "Injection". I would imagine, in this declarative model, that a user has more control over the definition of a stage/configuration/module. Do you have a concrete example of how this would work ?